python/tests/test_zarr.py

```diff
@@ -10,58 +10,129 @@
 zarr = pytest.importorskip("zarr")
 kvikio_zarr = pytest.importorskip("kvikio.zarr")
 
-# To support CuPy arrays, we need the `meta_array` argument introduced in
-# Zarr v2.13, see <https://github.com/zarr-developers/zarr-python/pull/934>
-if not hasattr(zarr.Array, "meta_array"):
-    pytest.skip("requires Zarr v2.13+", allow_module_level=True)
+
+if not kvikio_zarr.supported():
+    pytest.skip("requires Zarr v2.15+", allow_module_level=True)
 
 
 @pytest.fixture
 def store(tmp_path):
     """Fixture that creates a GDS Store"""
+    cupy.arange(1)  # Making sure that CUDA has been initialized
     return kvikio_zarr.GDSStore(tmp_path / "test-file.zarr")
 
 
-@pytest.mark.parametrize("array_type", ["numpy", "cupy"])
-def test_direct_store_access(store, array_type):
+def test_direct_store_access(store, xp):
     """Test accessing the GDS Store directly"""
 
-    module = pytest.importorskip(array_type)
-    a = module.arange(5, dtype="u1")
+    a = xp.arange(5, dtype="u1")
     store["a"] = a
     b = store["a"]
 
-    # Notice, GDSStore always returns a cupy array
-    assert type(b) is cupy.ndarray
-    cupy.testing.assert_array_equal(a, b)
+    # Notice, when not using getitems(), GDSStore returns bytes always
+    assert isinstance(b, bytes)
+    a.data == b
```

Comment:
This equality test doesn't do anything (since there is no assertion).
```suggestion
    assert (xp.frombuffer(b, dtype="u1") == a).all()
```
?